src/transfers-history/model/state.ts

```diff
@@ -0,0 +1,102 @@
+import { BigNumberish } from "ethers";
+import { ChainId } from "../../constants";
+
+export type TransferStatus = "pending" | "filled";
+
+export type Transfer = {
+  depositId: number;
+  depositTime: number;
+  status: TransferStatus;
+  filled: BigNumberish;
+  sourceChainId: ChainId;
+  destinationChainId: number;
+  assetAddr: string;
+  amount: BigNumberish;
+  depositTxHash: string;
+};
+
+export type TransferFilters = {
+  address?: string;
+  status: TransferStatus;
+};
+
+type QueryProgress = { latestFromBlock?: number; latestToBlock?: number; latestFromTimestamp?: number };
+type ChainQueryProgress = Record<number, QueryProgress>;
+
+type ChainTransfers = {
+  [chainId: number]: {
+    [depositId: number]: Transfer;
+  };
+};
+
+type BlockLowerBound = {
+  [chainId: number]: number;
+};
+
+const blockLowerBoundInitialValues = {
+  [ChainId.MAINNET]: 0,
+  [ChainId.RINKEBY]: 0,
+  [ChainId.KOVAN]: 0,
+  [ChainId.OPTIMISM_KOVAN]: 0,
+  [ChainId.ARBITRUM_RINKEBY]: 9828565,
+};
+
+export class State {
+  public constructor(
+    public completedTransfers: Transfer[] = [],
+    public pendingTransfers: Transfer[] = [],
+    public filters: TransferFilters | undefined = undefined,
+    public progress: ChainQueryProgress = {},
+    public chainTransfers: ChainTransfers = {},
+    public blockLowerBound: BlockLowerBound = blockLowerBoundInitialValues
+  ) {}
+
+  public clean() {
+    this.completedTransfers = [];
+    this.pendingTransfers = [];
+    this.filters = undefined;
+    this.progress = {};
+    this.chainTransfers = {};
+  }
+
+  public insertTransfer(chainId: number, depositId: number, transfer: Transfer) {
+    if (!this.chainTransfers[chainId]) {
+      this.chainTransfers[chainId] = {};
+    }
+    this.chainTransfers[chainId][depositId] = transfer;
+  }
+
+  public insertQueryProgress(chainId: ChainId, progress: QueryProgress) {
+    this.progress[chainId] = progress;
+  }
+
+  public setLatestFromTimestamp(chainId: ChainId, from: number) {
+    if (!this.progress[chainId]) {
+      this.progress[chainId] = {};
+    }
```

Comment:
Do the chainIds used in TransfersHistoryClient ever change? Any reason we can't make those constructor params here? If so, we could pre-initialize all of these objects (and maybe the values too) so we wouldn't need to do this check over and over.